test/replaytests/replay_test.go

```diff
@@ -311,8 +311,17 @@ func (s *replayTestSuite) TestUnkownSDKFlag() {
 func (s *replayTestSuite) TestUpdateWorkflow() {
 	replayer := worker.NewWorkflowReplayer()
 	replayer.RegisterWorkflow(UpdateWorkflow)
-	err := replayer.ReplayWorkflowHistoryFromJSONFile(ilog.NewDefaultLogger(), "update.json")
-	require.NoError(s.T(), err)
+
+	for _, name := range [...]string{
+		"update_before_protocol_message_command",
+		"update_with_protocol_message_command",
+	} {
+		s.Run(name, func() {
+			err := replayer.ReplayWorkflowHistoryFromJSONFile(ilog.NewDefaultLogger(), name+".json")
```

Comment:
can we add some tests for failure cases? Another integration test case I would love to see is accepting, completing and accepting+completing an update in the last workflow task of a history and verifying the commands are present in history. Currently all those situations are broken and I was expecting `ProtocolMessageCommands` to fix them.

Reply:
I can do that. FWIW that was fixed even before ProtocoMessageCommand went in on the server. There _was_ a time when it was broken though - you didn't hallucinate that.

Reply:
I added replay tests for complete and accept+complete in the last WFT. I'll add a test for a non-deterministic update implementation ... can you think of anything else apart from very obviously broken things like events in the wrong place that would get caught in `features`?

Reply:
>can you think of anything else apart from very obviously broken things like events in the wrong place that would get caught in features

not sure I understand the question? could you rephrase it?

Reply:
Basically just trying to brainstorm negative test scenarios beyond non-determinism.

Reply:
No other then trying different events in the update handler to cause non determinism